util/encoding/key_encoding.go

```diff
@@ -410,44 +454,119 @@ func floatMandE(f float64) (int, []byte) {
 	if f < 0 {
 		f = -f
 	}
-	i := 0
-	if f >= 1 {
-		for t := f; t >= 1; t /= 100 {
-			i++
-		}
+
+	// Use strconv.FormatFloat to handle the intricacies of determining how much
+	// precision is necessary to precisely represent f. The 'e' format is
+	// d.dddde±dd.
+	b := strconv.AppendFloat(nil, f, 'e', -1, 64)
+	if len(b) < 5 {
```

Comment:
If zero-padding to two digit exponents is not guaranteed (and I don't see it in the docs), then 4-byte strings are possible (and I think the code that follows would work with them).